pkg/sparrow/run.go

```diff
@@ -131,25 +138,37 @@ func (s *Sparrow) ReconcileChecks(ctx context.Context) {
 			// TODO discuss whether this should return an error instead?
 			continue
 		}
-		check.RegisterHandler(ctx, &s.routingTree)
-		go check.Run(ctx)
+		check.RegisterHandler(ctx, s.routingTree)
+		go func() {
+			err := check.Run(ctx)
+			if err != nil {
+				log.ErrorContext(ctx, "Failed to run check", "name", name, "error", err)
+				// TODO: anything else to do here?
+			}
+		}()
 	}
 
 	for existingCheckName, existingCheck := range s.checks {
 		// Check has been removed from config; shutdown and remove
-		if _, ok := s.cfg.Checks[existingCheckName]; !ok {
-			existingCheck.DeregisterHandler(ctx, &s.routingTree)
-			existingCheck.Shutdown(ctx)
-			if c, ok := s.resultFanIn[existingCheckName]; ok {
-				// close fan in channel if it exists
-				close(c)
-				delete(s.resultFanIn, existingCheckName)
-			}
+		log := logger.FromContext(ctx).With("checkName", existingCheckName)
+		if _, ok := s.cfg.Checks[existingCheckName]; ok {
+			continue
+		}
 
-			delete(s.checks, existingCheckName)
+		existingCheck.DeregisterHandler(ctx, s.routingTree)
+		err := existingCheck.Shutdown(ctx)
+		if err != nil {
+			log.ErrorContext(ctx, "Failed to shutdown check", "error", err)
+			// TODO: discuss what should happen
+		}
+		if c, ok := s.resultFanIn[existingCheckName]; ok {
+			// close fan in the channel if it exists
+			close(c)
+			delete(s.resultFanIn, existingCheckName)
 		}
```

Comment:
In my opinion these variable names are still way too long, why not `name`, `check` or even `n`, `c`?

Reply:
You've refactored the function so we'll tackle that there